sys/random/Makefile

```diff
@@ -11,4 +14,28 @@ ifneq (,$(filter prng_tinymt32,$(USEMODULE)))
   DIRS += tinymt32
 endif
 
+ifeq (,$(RIOT_CI_BUILD))
+  # random seed value newly generated on each build
+  RANDOM_SEED := $(shell $(RIOTTOOLS)/randhex/randhex.py 32)
+else
+  # hardcoded canary value to not break CI test cache
+  RANDOM_SEED := 0x1
+endif
+
+HAS_RNG := 0
+ifneq (,$(filter puf_sram,$(USEMODULE)))
+  HAS_RNG = 1
+endif
+ifneq (,$(filter periph_hwrng,$(FEATURES_PROVIDED)))
+  HAS_RNG = 1
+endif
+ifneq (,$(filter periph_cpuid,$(FEATURES_PROVIDED)))
+  HAS_RNG = 1
+endif
```

Comment:
That looks bad although I believe there is currently no nicer way. @leandrolanzieri can you have a look?

Reply:
Also fails the `buildsystem_sanity_check`:

	Invalid build system patterns found by ./dist/tools/buildsystem_sanity_check/check.sh:
	Modules should not check the content of FEATURES_PROVIDED/_REQUIRED/OPTIONAL
		sys/random/Makefile:ifneq (,$(filter periph_hwrng,$(FEATURES_PROVIDED)))
		sys/random/Makefile:ifneq (,$(filter periph_cpuid,$(FEATURES_PROVIDED)))

Reply:
One should check for `FEATURES_USED`